climaf/functions.py

```diff
@@ -439,12 +470,130 @@ def zonmean_interpolation(dat1,dat2=None,vertical_levels=None,cdo_horizontal_gri
+    it is converted to Pa if necessary directly in the netcdf file(s) corresponding to dat1(2).
+
+       >>> dat = ds(project='CMIP5',model='IPSL-CM5A-LR',variable='ua',period='1980-1985',
+                    experiment='historical',table='Amon')
+       >>> ref = ds(project='ref_pcmdi',variable='ua',product='ERAINT')
+
+       >>> zonmean_dat = zonmean(time_average(dat))
+       >>> zonmean_ref = zonmean(time_average(ref))
+
+       >>> dat_interpolated_on_ref = zonmean_interpolation(zonmean_dat,zonmean_ref)
+       >>> dat_interpolated_on_list_of_levels = zonmean_interpolation(zonmean_dat,vertical_levels='100000,85000,50000,20000,10000,5000,2000,1000')
+
+    """
+
+    from climaf.anynetcdf import ncf
+
+    file1 = cfile(dat1)
+    ncfile1 = ncf(file1)
+
+    # -- First, we check the unit of the vertical dimension of file1
+    levname1=None
+    for varname in ncfile1.variables:
+        if varname.lower() in ['level','levels','lev','levs','depth','deptht','plev'] or 'plev' in varname.lower():
+            levname1=varname
+    if not levname1:
+        print 'Name of the vertical axis not found for dat1'
+    levunits1 = ncfile1.variables[levname1].units
+    if levunits1.lower() in ['hpa','millibar','mbar','hectopascal']:
+        # -- Multiplier par 100
+        cmd1 = 'ncap2 -As "'+levname1+'='+levname1+'*100" '+file1+' '+file1
+        cmd2 = 'ncatted -O -a units,'+levname1+',o,c,Pa '+file1
+        print cmd1
+        print cmd2
+        os.system(cmd1)
+        os.system(cmd2)
+    # -> The vertical axis of file1 is now set to Pa
+    #
+    # -- Second, we check the unit of the vertical dimension of file2
+    if dat2:
+        file2 = cfile(dat2)
+        print 'file2 = ',file2
+        ncfile2 = ncf(file2)
+
+        levname2=None
+        for varname in ncfile2.variables:
+            if varname.lower() in ['level','levels','lev','levs','depth','deptht','plev'] or 'plev' in varname.lower():
+                levname2=varname
+        print 'levname2 = ',levname2
+        if not levname2:
+            print 'Name of the vertical axis not found for dat2'
+        levunits2  = ncfile2.variables[levname2].units
+        print 'ncfile2 = ',ncfile2
+        #print 'dir(ncfile2.variables[levname2]) = ',dir(ncfile2.variables[levname2])
+        #print 'ncfile2.variables[levname2].getValue() = ',ncfile2.variables[levname2].getValue()
+        try:
+           levValues2 = ncfile2.variables[levname2].getValue()
+        except:
+           try:
+              levValues2 = ncfile2.variables[levname2].data
+           except:
+              levValues2 = ncfile2[levname2][0:len(ncfile2[levname2])]
+        if levunits2.lower() in ['hpa','millibar','mbar','hectopascal']:
+            # -- Multiplier par 100
+            cmd1 = 'ncap2 -As "'+levname2+'='+levname2+'*100" '+file2+' '+file2
+            cmd2 = 'ncatted -O -a units,'+levname2+',o,c,Pa '+file2
+            print cmd1
+            print cmd2
+            os.system(cmd1)
```

Comment:
Instead of doing this step, I now declare a new CliMAF operator that does this sequence of nco commands. This way the corrected file is a new file, with its complete story.